diff --git a/src/create-module-declaration.js b/src/create-module-declaration.js
index 8cbefe79df2d126866843d18f94397fb60d6b0c1..b0abd63a5f37500aaebb64e434342c56534a13f0 100644
--- a/src/create-module-declaration.js
+++ b/src/create-module-declaration.js
@@ -246,7 +246,7 @@ export function create_module_declaration(id, entry, created, resolve, options)
 				walk(module.ast, (node) => {
 					if (is_reference(node) && ts.isQualifiedName(node.parent)) {
 						const binding = module.import_all.get(node.getText(module.ast));
-						if (binding) {
+						if (binding && !binding.external) {
 							result.remove(node.pos, result.original.indexOf('.', node.end) + 1);
 							const declaration = bundle
 								.get(binding.id)
